android/src/main/java/com/swmansion/gesturehandler/core/FlingGestureHandler.kt

```diff
@@ -3,15 +3,63 @@ package com.swmansion.gesturehandler.core
 import android.os.Handler
 import android.os.Looper
 import android.view.MotionEvent
+import android.view.VelocityTracker
+import com.swmansion.gesturehandler.core.GestureHandler.Companion
+import kotlin.math.hypot
+
+class Vector {
+  var x: Double = 0.0
+  var y: Double = 0.0
+  private var uX: Double = 0.0
+  private var uY: Double = 0.0
+
+  fun fromDirection(direction: Int) = also {
+    val (x, y) = when (direction) {
+      Companion.DIRECTION_LEFT -> Pair(-1.0, 0.0)
+      Companion.DIRECTION_RIGHT -> Pair(1.0, 0.0)
+      Companion.DIRECTION_UP -> Pair(0.0, -1.0)
+      Companion.DIRECTION_DOWN -> Pair(0.0, 1.0)
+      else -> Pair(0.0, 0.0)
+    }
+
+    this.x = x
+    this.uX = x
+    this.y = y
+    this.uY = y
+  }
+
+  fun fromVelocity(tracker: VelocityTracker) = also {
+    tracker.computeCurrentVelocity(1000)
+
+    this.x = tracker.xVelocity.toDouble()
+    this.y = tracker.yVelocity.toDouble()
+
+    val magnitude = hypot(this.x, this.y)
+    if (magnitude < 0.001) {
+      this.uX = 0.0
+      this.uY = 0.0
+    }
+
+    this.uX = this.x / magnitude
+    this.uY = this.y / magnitude
+  }
+
+  fun computeSimilarity(vector: Vector): Double {
+    return this.uX * vector.uX + this.uY * vector.uY
+  }
+
+  fun computeMagnitude(): Double {
+    return hypot(this.x, this.y)
+  }
```

Comment:
We can make this into a read-only property
```suggestion
  val magnitude: Double
    get() = hypot(this.x, this.y)
```